evm_arithmetization/src/cpu/kernel/asm/main.asm

```diff
@@ -144,10 +158,25 @@ global perform_final_checks:
     // stack: cum_gas, txn_counter, num_nibbles, txn_nb
     // Check that we end up with the correct `cum_gas`, `txn_nb` and bloom filter.
     %mload_global_metadata(@GLOBAL_METADATA_BLOCK_GAS_USED_AFTER) %assert_eq
-    DUP3 %mload_global_metadata(@GLOBAL_METADATA_TXN_NUMBER_AFTER) %assert_eq
+    DUP3
+    %mload_global_metadata(@GLOBAL_METADATA_TXN_NUMBER_AFTER) %assert_eq
     %pop3
+
+    PROVER_INPUT(trie_ptr::state)
+
+    %mstore_global_metadata(@GLOBAL_METADATA_STATE_TRIE_ROOT)
+    %set_initial_tries
```

Comment:
Maybe we can make use of this call to validate the initial value of `@GLOBAL_METADATA_ACCOUNTS_LINKED_LIST_LEN`, if we store it somewhere?

Reply:
Or maybe it would be easier to check it in `store_initial_accounts` (resp. `store_initial_slots) since we're iterating over the accounts (resp. slots) there?

Reply:
I removed the PROVER_INPUT calls and computed the lengths in `store_initial_accounts` and `store_initial_slots` directly